### DISCUSSION TITLE
Is Databend's execution model better than the volcano model used by Datafusion?

### DISCUSSION THREAD

**Discussion by JasonLi-cn** [databendlabs/databend#7954] [Q&A] [answered]
I understand that the volcano model has some obvious drawbacks, such as large number of virtual function calls, conditional branch prediction failures, and so on, so Databend doesn't use it, but I'm not sure what the advantages of the Databend execution model are right now, so please help me introduce it. I have read [this article](https://bohutang.me/2020/06/11/clickhouse-and-friends-processor/), but I still don't quite understand the advantages of DAG Scheduler. 
Thanks~

**Comment by xudong963** [comment, marked as answer]
Datafusion is iterating on a new execution model, which you can learn more about in this document: https://docs.google.com/document/d/1txX60thXn1tQO1ENNT8rwfU3cXLofa7ZccnvP4jD6AA/edit#heading=h.3iwlbn2gzs29

The new execution model has many similarities to the current execution model of databend (both largely avoiding the drawbacks of the valcano model)

If you want to know more about the advantages of DAG Scheduler in databend, I'll ping @zhang2014  for help :)

**Reply by JasonLi-cn** [reply]
Thanks @xudong963. 
Could you please introduce the advantages of DAG Scheduler in databend? @zhang2014
